docs/modeling/example-fitting-model-sets.rst

```diff
@@ -1,37 +1,120 @@
 Fitting Model Sets
 ==================
 
-Fitting a polynomial model to multiple data sets simultaneously::
+Astropy model sets let you fit the same (linear) model to lots of independent
+data sets. It does the looping for you, and can be significantly faster than
+doing the looping yourself. But getting the data into the right shape can be 
+a bit tricky. 
 
-      >>> from astropy.modeling import models, fitting
-      >>> import numpy as np
-      >>> p1 = models.Polynomial1D(3)
-      >>> p1.c0 = 1
-      >>> p1.c1 = 2
-      >>> print(p1)
-      Model: Polynomial1D
-      Inputs: ('x',)
-      Outputs: ('y',)
-      Model set size: 1
-      Degree: 3
-      Parameters:
-           c0  c1  c2  c3
-          --- --- --- ---
-          1.0 2.0 0.0 0.0
-      >>> x = np.arange(10)
-      >>> y = p1(x)
-      >>> yy = np.array([y, y])
-      >>> p2 = models.Polynomial1D(3, n_models=2)
-      >>> pfit = fitting.LinearLSQFitter()
-      >>> new_model = pfit(p2, x, yy)
-      >>> print(new_model)  # doctest: +SKIP
-      Model: Polynomial1D
-      Inputs: 1
-      Outputs: 1
-      Model set size: 2
-      Degree: 3
-      Parameters:
-           c0  c1         c2                 c3
-          --- --- ------------------ -----------------
-          1.0 2.0 -5.86673908219e-16 3.61636197841e-17
-          1.0 2.0 -5.86673908219e-16 3.61636197841e-17
+In the example below, we create a 3D data cube where the first dimension is a ramp, 
+for example as from non-destructive readouts of an IR detector. So each pixel has a 
+depth along a time axis, and flux that results a total number of counts that is 
+increasing with time. We will be fitting a 1D polynomial vs. time to estimate the 
+flux in counts/second (the slope of the fit).
+
+   .. plot::
+       :include-source:
+
+       import numpy as np
+       import matplotlib.pyplot as plt
+       from scipy import stats
+       from astropy.modeling import models, fitting
+       
+       # Set up the shape of the image and create the time axis
+       DEPTH,WIDTH,HEIGHT=10,101,102 # Time is along the DEPTH axis
+       t = np.arange(DEPTH,dtype=np.float64)*10. # e.g. readouts every 10 seconds
+       
+       # Make up a flux in each pixel 
+       fluxes = np.arange(1.*WIDTH*HEIGHT).reshape(HEIGHT,WIDTH)
+       # Create the ramps by integrating the fluxes along the time steps
+       image = np.stack([fluxes*dt for dt in t],axis=0)
+       # Add some Gaussian noise to each sample
+       image += stats.norm.rvs(0.,image*0.05,size=image.shape) # Add noise
+       print("The image shape is: ",image.shape)
```

Comment:
The print output cannot be doctested the way it is currently set up. I recommend refactoring such that it can be doctested, or just remove all the `print()`.

Reply:
I couldn't figure out how to get printing to work from within a .plot directive. Do you have suggestions for how to refactor to get this to work?

Reply:
Can we just outright remove the `print`s? If the plot renders, we know the example works, isn't that enough?

Reply:
One way to have a narrative flow of code snippets and then have the plot without `:include-source:`. That would unfortunately double the lines, but the rendered version will be much nicer with a logical flow and outputs that can be doctested.

Reply:
Refactored along the lines. I don't know how to tell if the tests are running and passing on the code snippets though. Also, it would be nice to give the user the option of downloading the entire source rather than the snippets, and I'm not sure how to do that.

Reply:
Yes, as I recall we don't have a solution for downloading the figure source codes yet. Maybe this can be another way to extend the concept of examples? 
cc @kelle @jonathansick